pkg/virt-launcher/virtwrap/network/network_test.go

```diff
@@ -32,6 +32,29 @@ var _ = Describe("VMNetworkConfigurator", func() {
 	Context("interface configuration", func() {
 		var launcherPID *int
 
+		Context("when vm has no network source", func() {
+			var (
+				vmi                   *v1.VirtualMachineInstance
+				vmNetworkConfigurator *VMNetworkConfigurator
+			)
+			BeforeEach(func() {
+				vmi = newVMIBridgeInterface("testnamespace", "testVmName")
+				vmi.Spec.Networks = []v1.Network{{
+					Name:          "default",
+					NetworkSource: v1.NetworkSource{},
+				}}
+				vmNetworkConfigurator = NewVMNetworkConfigurator(vmi, fake.NewFakeInMemoryNetworkCacheFactory())
+			})
+			It("should propagate errors when phase1 is called", func() {
+				err := vmNetworkConfigurator.SetupPodNetworkPhase1(0)
+				Expect(err).To(MatchError("Network not implemented"))
+			})
+			It("should propagate errors when phase2 is called", func() {
+				err := vmNetworkConfigurator.SetupPodNetworkPhase2(nil)
+				Expect(err).To(MatchError("Network not implemented"))
```

Comment:
I'd make this a one liner, and simply use `HaveOccurred()`. This makes it look like you're looking for particular error types - I guess this one was simply the simplest way you found to make this go kaboom.

While you're at it, please also set the domain xml in a variable to improve the readability.

Reply:
Well I want to be sure that what I am propagating is from getNICs at the end is what I am testing, to perfect solution would be to mock getNICs but I don't thing it's worth it, If I use a HaveOcurred here I don't know if it's from getNICs or not.